Bottom: 3ca32272093f434912b043a1165fd2062df8772e
Top:    cfc773fb1984c9ad2d64ab134d2e66f663e3bb4e
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-04 18:15:39 +0800

Refresh of autofs-5.0.7-syncronize-handle_mounts-shutdown.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index 03fd266..6810c73 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1545,7 +1545,6 @@ static void handle_mounts_cleanup(void *arg)
 		if (!fc.busy)
 			pthread_kill(state_mach_thid, SIGTERM);
 		error(LOGOPT_ANY, "before signal fc.busy %d", fc.busy);
-		fc.busy++;
 		master_mutex_unlock();
 		handle_mounts_finish();
 		error(LOGOPT_ANY, "after wait fc.busy %d", fc.busy);
diff --git a/lib/master.c b/lib/master.c
index ece5afa..ba9ca75 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -746,6 +746,7 @@ void master_remove_mapent(struct master_mapent *entry)
 	if (!list_empty(&entry->list)) {
 		list_del_init(&entry->list);
 		list_add(&entry->join, &master->completed);
+		fc.busy++;
 	}
 
 	return;
@@ -1410,9 +1411,9 @@ void master_finish(struct master *master)
 		entry = list_entry(p, struct master_mapent, join);
 		p = p->next;
 		list_del(&entry->join);
+		fc.busy--;
 		master_free_mapent_sources(entry, 1);
 		master_free_mapent(entry);
-		fc.busy--;
 	}
 
 	status = pthread_cond_broadcast(&fc.cond);
